api/meetings/views.py

```diff
@@ -0,0 +1,240 @@
+
+from rest_framework import generics, permissions as drf_permissions
+from rest_framework.exceptions import NotFound
+from django.db.models import Q, Count, Subquery, OuterRef, Case, When, Value, CharField, F
+from django.db.models.functions import Length, Substr, Coalesce
+from django.contrib.contenttypes.models import ContentType
+
+from addons.osfstorage.models import OsfStorageFile
+from api.base import permissions as base_permissions
+from api.base.exceptions import InvalidFilterOperator
+from api.base.filters import ListFilterMixin
+from api.base.views import JSONAPIBaseView
+from api.base.utils import get_object_or_error
+from api.base.versioning import PrivateVersioning
+from api.meetings.serializers import MeetingSerializer, MeetingSubmissionSerializer
+from api.meetings.permissions import IsPublic
+from api.nodes.views import NodeMixin
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from osf.models import AbstractNode, Conference, Contributor, Tag, PageCounter
+from website import settings
+
+
+class MeetingMixin(object):
+    """Mixin with convenience method get_meeting
+    """
+
+    meeting_lookup_url_kwarg = 'meeting_id'
+
+    def get_meeting(self):
+        meeting = get_object_or_error(
+            Conference,
+            Q(endpoint=self.kwargs[self.meeting_lookup_url_kwarg]),
+            self.request,
+            display_name='meeting',
+        )
+        # caching num_submissions on the Conference object
+        meeting.num_submissions = meeting.submissions.count()
+        meeting.save()
+        return meeting
+
+
+class BaseMeetingView(JSONAPIBaseView, MeetingMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.MEETINGS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model = Conference
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+
+    serializer_class = MeetingSerializer
+    view_category = 'meetings'
+
+
+class MeetingList(BaseMeetingView, generics.ListAPIView, ListFilterMixin):
```

Comment:
Yes, I'm returning a queryset - Is it the name that's the issue?  `ListFilterMixin` is used for filtering querysets, did it used to be for literal lists? (I, too, took an extended break from backend work :)

Reply:
So, back in the day, we had ListFilterMixin and QueryFilterMixin (or similar). ListFilter was for views that returned lists (generally things that we had to add extra information to because ModularODM wasn't _quite_ as good as the Django ORM). It makes sense that we never return lists any more, but it's weird that they… oh, I see. They called it ListFilter because it's filtering a List view, not because it's filtering a python list. So this is literally only confusing to me. Mystery solved!